newrelic/newrelic.php

```diff
@@ -1,452 +1,647 @@
  *
  * @return bool
  */
-function newrelic_name_transaction($name) {}
+function newrelic_name_transaction(string $name): bool {}
 
 /**
- * Report an error at this line of code, with a complete stack trace.
+ * Use these calls to collect errors that the PHP agent does not collect automatically and to set the callback for
+ * your own error and exception handler.
+ *
+ * CAUTION: If you include an exception ($e), there are differences depending on the PHP version you are using:
+ * - PHP version 5 or lower: You must pass a valid PHP Exception class.
+ * - PHP version 7 or higher: You must pass a valid PHP Throwable interface.
+ *
+ * The PHP agent handles PHP errors and exceptions automatically for supported frameworks. If you want to collect
+ * errors that are not handled automatically so that you can query for those errors in New Relic and view error
+ * traces, you can use newrelic_notice_error.
+ *
+ * If you want to use your own error and exception handlers, you can set newrelic_notice_error as the callback.
+ *
+ * This function can handle a variable number of parameters. You can pass-in 1 or 5 parameters, depending on your use
+ * case:
+ * - newrelic_notice_error(string $message)
+ * - newrelic_notice_error(Throwable|Exception $e)
+ * - newrelic_notice_error(int $errno, string $errstr, string $errfile, int $errline, string $errcontext)
+ *
+ * @link  https://docs.newrelic.com/docs/agents/php-agent/php-agent-api/newrelic_notice_error/
+ * @since 2.6
+ *
+ * @param string|Throwable|Exception|int $messageOrExceptionOrCode Either an error message, an exception object, or
+ *                                                                 an
+ *                                                                 integer error code
+ * @param string|null                    $errstr                   [optional] Required if first argument is an
+ *                                                                 integer, Provide an error message that will be
+ *                                                                 meaningful to you when it displays in error
+ *                                                                 traces.
+ * @param string|null                    $errfile                  [optional] The name of the file that the error
+ *                                                                 occurred in.
+ * @param int|null                       $errline                  [optional] The line number where the error
+ *                                                                 occurred.
+ * @param string|null                    $errcontext               [optional] An array that points to the symbol
+ *                                                                 table
+ *                                                                 that was active when the error occurred.
+ *
+ * @return null
+ */
+function newrelic_notice_error(
+    string|Throwable|Exception|int $messageOrExceptionOrCode,
```

Comment:
Hmm..I don't think that joining several signatures into one will work well here because for example with such a signature it will be possible to make a call `newrelic_notice_error(1)` and IDE will not show any warning because such call will match this stub

Reply:
I agree, but I'm not sure PHPDoc has a good solution to this. It's basically IF firstParam IS int THEN secondParam IS Required. Does JetBrains have an internal PHP 8 Attribute that covers this?

Reply:
Unfortunately no, we don't have such an attribute yet. Why don't you like an approach with several signatures in this case?

Reply:
From the other side, this signature was also joined before and there were no any complains so probably we can live with such signature as well.